lnutil/msglib.go

```diff
@@ -1,47 +1,770 @@
+	MSGID_CLOSEREQ  = 0x20 // close channel
+	MSGID_CLOSERESP = 0x21
+
+	//Push Pull Messages
+	MSGID_DELTASIG  = 0x30 // pushing funds in channel; request to send
+	MSGID_SIGREV    = 0x31 // pulling funds; signing new state and revoking old
+	MSGID_GAPSIGREV = 0x32 // resolving collision
+	MSGID_REV       = 0x33 // pushing funds; revoking previous channel state
+
+	//not implemented
+	MSGID_FWDMSG     = 0x40
+	MSGID_FWDAUTHREQ = 0x41
+
+	//not implemented
+	MSGID_SELFPUSH = 0x50
+
+	//Tower Messages
+	MSGID_WATCH_DESC   = 0x60 // desc describes a new channel
+	MSGID_WATCH_COMMSG = 0x61 // commsg is a single state in the channel
+	MSGID_WATCH_DELETE = 0x62 // Watch_clear marks a channel as ok to delete.  No further updates possible.
+)
+
+//interface that all messages follow, for easy use
+type LitMsg interface {
+	Peer() uint32   //return PeerIdx
+	MsgType() uint8 //returns Message Type (see constants above)
+	Bytes() []byte  //returnns data of message as []byte with the MsgType() preceeding it
+}
+
+//method for finding what type of message a generic []byte is
+
+func LitMsgFromBytes(b []byte, peerid uint32) (LitMsg, error) {
+	msgType := b[0] // first byte signifies what type of message is
+
+	switch msgType {
+	case MSGID_TEXTCHAT:
+		return NewChatMsgFromBytes(b, peerid)
+	case MSGID_POINTREQ:
+		return NewPointReqMsgFromBytes(peerid)
+	case MSGID_POINTRESP:
+		return NewPointRespMsgFromBytes(b, peerid)
+	case MSGID_CHANDESC:
+		return NewChanDescMsgFromBytes(b, peerid)
+	case MSGID_CHANACK:
+		return NewChanAckMsgFromBytes(b, peerid)
+	case MSGID_SIGPROOF:
+		return NewSigProofMsgFromBytes(b, peerid)
+
+	case MSGID_CLOSEREQ:
+		return NewCloseReqMsgFromBytes(b, peerid)
+	/* not implemented
+	case MSGID_CLOSERESP:
+	*/
+
+	case MSGID_DELTASIG:
+		return NewDeltaSigMsgFromBytes(b, peerid)
+	case MSGID_SIGREV:
+		return NewSigRevFromBytes(b, peerid)
+	case MSGID_GAPSIGREV:
+		return NewGapSigRevFromBytes(b, peerid)
+	case MSGID_REV:
+		return NewRevMsgFromBytes(b, peerid)
+
+	/*
+		case MSGID_FWDMSG:
+		case MSGID_FWDAUTHREQ:
+
+		case MSGID_SELFPUSH:
+	*/
+
+	case MSGID_WATCH_DESC:
+		return NewWatchDescMsgFromBytes(b, peerid)
+	case MSGID_WATCH_COMMSG:
+		return NewComMsgFromBytes(b, peerid)
+	/*
+		case MSGID_WATCH_DELETE:
+	*/
+
+	default:
+		return nil, fmt.Errorf("Unknown message of type %d ", msgType)
+	}
+}
+
+//----------
+
+//text message
+type ChatMsg struct {
 	PeerIdx uint32
-	ChanIdx uint32 // optional, may be 0
-	MsgType uint8
-	Data    []byte
+	Text    string
 }
 
-const (
-	MSGID_POINTREQ  = 0x30
-	MSGID_POINTRESP = 0x31
-	MSGID_CHANDESC  = 0x32
-	MSGID_CHANACK   = 0x33
-	MSGID_SIGPROOF  = 0x34
+func NewChatMsg(peerid uint32, text string) *ChatMsg {
+	t := new(ChatMsg)
+	t.PeerIdx = peerid
+	t.Text = text
+	return t
+}
 
-	MSGID_CLOSEREQ  = 0x40 // close channel
-	MSGID_CLOSERESP = 0x41
+func NewChatMsgFromBytes(b []byte, peerid uint32) (*ChatMsg, error) {
+	c := new(ChatMsg)
+	c.PeerIdx = peerid
+	b = b[1:]
```

Comment:
*almost* no way to error, but if b is 0 (or 1) bytes long this will crash the program; put in a
```
if len(b) < 2 { return nil, fmt.Errorf("got %d bytes, expect 2 or more", len(b)) }
```
or equiv